pkg/preflight/aws.go

```diff
@@ -37,29 +38,26 @@ var (
 
 type aws struct {
 	checker
-	ops  cloudops.Ops
-	zone string
+	ops            cloudops.Ops
+	zone           string
+	credentialHash string
 }
 
 func (a *aws) initCloudOps(cluster *corev1.StorageCluster) error {
 	// Set env vars for user provided credentials: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
-	existingAccessKey := os.Getenv(awsAccessKeyEnvName)
-	existingSecretKey := os.Getenv(awsSecretKeyEnvName)
+	// only set env vars when client init is needed
 	if err := a.setAWSCredentialEnvVars(cluster); err != nil {
+		resetAWSCredentialEnvVars()
 		return err
 	}
 
-	// init aws client when
-	// 1. it's never created
-	// 2. user provided credentials got changed
-	if a.ops != nil &&
-		existingAccessKey == os.Getenv(awsAccessKeyEnvName) &&
-		existingSecretKey == os.Getenv(awsSecretKeyEnvName) {
+	if a.ops != nil {
```

Comment:
why dont compare the value here? if the value changes the client wont be recreated?

Reply:
now all that is handled in `setAWSCredentialEnvVars`, the env vars are only set when it needs to create a new client, e.g. credentials removed, added or updated, otherwise we don't set the env.